src/main/java/uk/co/real_logic/benchmarks/rtt/Configuration.java

```diff
@@ -0,0 +1,543 @@
+            this.senderIdleStrategy = senderIdleStrategy;
+            return this;
+        }
+
+        /**
+         * Set the {@link IdleStrategy} to use when receiving messages.
+         *
+         * @param receiverIdleStrategy idle strategy to use for receiving messages.
+         * @return this for a fluent API.
+         */
+        public Builder receiverIdleStrategy(final IdleStrategy receiverIdleStrategy)
+        {
+            this.receiverIdleStrategy = receiverIdleStrategy;
+            return this;
+        }
+
+        /**
+         * Create a new instance of {@link Configuration} class from this builder.
+         *
+         * @return a {@link Configuration} instance
+         */
+        public Configuration build()
+        {
+            return new Configuration(this);
+        }
+    }
+
+    /**
+     * Create a {@link Configuration} instance based on the provided system properties.
+     *
+     * @param properties system properties.
+     * @return a {@link Configuration} instance.
+     * @throws NullPointerException if {@code properties == null}
+     */
+    public static Configuration fromProperties(final Map<String, String> properties)
+    {
+        requireNonNull(properties);
+
+        final Builder builder = new Builder();
+        if (isPropertyProvided(properties, WARM_UP_ITERATIONS_PROP_NAME))
+        {
+            builder.warmUpIterations(parseInt(properties, WARM_UP_ITERATIONS_PROP_NAME));
+        }
+        if (isPropertyProvided(properties, WARM_UP_MESSAGES_PROP_NAME))
+        {
+            builder.warmUpNumberOfMessages(parseInt(properties, WARM_UP_MESSAGES_PROP_NAME));
+        }
+        if (isPropertyProvided(properties, WARM_UP_MESSAGES_PROP_NAME))
+        {
+            builder.warmUpNumberOfMessages(parseInt(properties, WARM_UP_MESSAGES_PROP_NAME));
+        }
+        if (isPropertyProvided(properties, ITERATIONS_PROP_NAME))
+        {
+            builder.iterations(parseInt(properties, ITERATIONS_PROP_NAME));
+        }
+        if (isPropertyProvided(properties, BURST_SIZE_PROP_NAME))
+        {
+            builder.burstSize(parseInt(properties, BURST_SIZE_PROP_NAME));
+        }
+        if (isPropertyProvided(properties, MESSAGE_SIZE_PROP_NAME))
+        {
+            builder.messageSize(parseInt(properties, MESSAGE_SIZE_PROP_NAME));
+        }
+        if (isPropertyProvided(properties, SENDER_IDLE_STRATEGY_PROP_NAME))
+        {
+            builder.senderIdleStrategy(parseIdleStrategy(properties, SENDER_IDLE_STRATEGY_PROP_NAME));
+        }
+        if (isPropertyProvided(properties, RECEIVER_IDLE_STRATEGY_PROP_NAME))
+        {
+            builder.receiverIdleStrategy(parseIdleStrategy(properties, RECEIVER_IDLE_STRATEGY_PROP_NAME));
+        }
+        builder.numberOfMessages(parseInt(properties, MESSAGES_PROP_NAME));
+        builder.messageProviderClass(parseClass(properties, MESSAGE_PROVIDER_PROP_NAME, MessageProvider.class));
+        return builder.build();
+    }
```

Comment:
It is difficult for a dyslexic like me to read code that does not use whitespace for separation.

Reply:
Will cleanup.